python/tank/platform/bundle.py

```diff
@@ -804,28 +804,57 @@ def __post_process_settings_r(self, key, value, schema):
         
         return processed_val
         
-    def __resolve_setting_value(self, key, value):
+    def __resolve_setting_value(self, settings, key, default):
         """
         Resolve a setting value.  Exposed to allow values
         to be resolved for settings derived outside of the 
         app.
-        
-        :param key:   setting name
-        :param value: setting value
+
+        :param settings: the settings source
+        :param key: setting name
+        :param default: a default value to use for the setting
         """
-        if value is None:
-            return value
-        
-        # try to get the type for the setting
-        # (may fail if the key does not exist in the schema,
-        # which is an old use case we need to support now...)
+
+        # start with the supplied default value in case we need to short circuit
+        # before checking against the schema
+        value = default
+
+        # attempt to get the value from the settings.
+        if key in settings:
+            value = settings[key]
+
+        # try to get the type for the setting (may fail if the key does not
+        # exist in the schema. an old use case we need to support now...)
         try:
             schema = self.__descriptor.get_configuration_schema().get(key)
         except:
             schema = None
-        
-        if schema:
-            # post process against schema
-            value = self.__post_process_settings_r(key, value, schema)
-            
-        return value
+
+        if not schema:
+            # No schema, can't post process or retrieve a default. Just return
+            # the value we have.
+            return value
+
+        # Schema exists. If the key wasn't in the settings, fallback to the
+        # default value in the manifest.
+        if key not in settings:
+
+            if schema.get("type") == "hook":
```

Comment:
Could this value-resolve logic be broken out into a method and reused? It seems to be in two places and with the more complex logic for defaults (default_value_tk-maya), it will grow in size and complexity.